src/euphorie/upgrade/deployment/v18/20250521110138_fix_default_language/upgrade.py

```diff
@@ -0,0 +1,28 @@
+from ftw.upgrade import UpgradeStep
+from plone import api
+
+
+class FixDefaultLanguage(UpgradeStep):
+    """Fix default language."""
+
+    def __call__(self):
+        default_language = api.portal.get_registry_record("plone.default_language")
+        available_languages = api.portal.get_registry_record(
+            "plone.available_languages"
+        )
+        if default_language not in available_languages:
+
+            if "en" in available_languages:
+                # Set the default language to English if it's available
+                api.portal.set_registry_record("plone.default_language", "en")
+            elif available_languages and len(available_languages) == 1:
+                # Set the default language to the first available language
```

Comment:
The comment says “first available language” but the check is `== 1`, so it's the *only* available language. Is that what's intended, or should it be `>= 1`?

Reply:
If we have more than one language, the default language cannot be set with the risk of annoying somebody, for this reason I am raising a `ValueError`.
I will update the comment.